autogpt/setup.py

```diff
@@ -74,97 +75,260 @@ def prompt_user() -> AIConfig:
     logger.typewriter_log(
         f"{ai_name} here!", Fore.LIGHTBLUE_EX, "I am at your service.", speak_text=True
     )
 
-    # Get AI Role from User
-    logger.typewriter_log(
-        "Describe your AI's role: ",
-        Fore.GREEN,
-        "For example, 'an AI designed to autonomously develop and run businesses with"
-        " the sole goal of increasing your net worth.'",
-    )
-    ai_role = utils.clean_input(f"{ai_name} is: ")
-    if ai_role == "":
-        ai_role = "an AI designed to autonomously develop and run businesses with the"
-        " sole goal of increasing your net worth."
+    # Handle ai_role: str
+    if editing:
+        logger.typewriter_log(
+            "Describe your AI's role:",
+            Fore.GREEN,
+            f"Current: '{config.ai_role}', use [Enter] to keep the current role / save the input.",
+            speak_text=True,
+        )
+    else:
+        logger.typewriter_log(
+            "Describe your AI's role:",
+            Fore.GREEN,
+            "For example, 'an AI designed to autonomously develop and run businesses with the sole goal of increasing your net worth.'",
+            speak_text=True,
+        )
 
-    # Enter up to 5 goals for the AI
-    logger.typewriter_log(
-        "Enter up to 5 goals for your AI: ",
-        Fore.GREEN,
-        "For example: \nIncrease net worth, Grow Twitter Account, Develop and manage"
-        " multiple businesses autonomously'",
+    ai_role = utils.clean_input(f"{ai_name} is: ") or (
+        config.ai_role
+        if config is not None
+        else "an AI designed to autonomously develop and run businesses with the sole goal of increasing your net worth."
     )
-    logger.info("Enter nothing to load defaults, enter nothing when finished.")
-    ai_goals = []
-    for i in range(5):
-        ai_goal = utils.clean_input(f"{Fore.LIGHTBLUE_EX}Goal{Style.RESET_ALL} {i+1}: ")
-        if ai_goal == "":
-            break
-        ai_goals.append(ai_goal)
-    if not ai_goals:
-        ai_goals = [
-            "Increase net worth",
-            "Grow Twitter Account",
-            "Develop and manage multiple businesses autonomously",
-        ]
-
-    # Get API Budget from User
-    logger.typewriter_log(
-        "Enter your budget for API calls: ",
-        Fore.GREEN,
-        "For example: $1.50",
-    )
-    logger.info("Enter nothing to let the AI run without monetary limit")
+
+    # Handle ai_goals: list
+    if editing and max_goals:
+        # Edit existing goals
+        default_goals = config.ai_goals if config else []
+        if editing:
+            logger.typewriter_log(
+                f"Enter up to {max_goals} goals for your AI:",
+                Fore.GREEN,
+                "use [Enter] to keep the current goal / save the input.",
+                speak_text=True,
+            )
+        ai_goals = list(default_goals)
+        kept_goals = []
+        deleted_goals_indices = []
+
+        for i in range(len(ai_goals)):
```

Comment:
Seems easier to just do

```suggestion
        for ai_goal in ai_goals:
```